limitador/src/storage/redis/counters_cache.rs

```diff
@@ -260,30 +245,24 @@ impl CountersCache {
         counter: Counter,
         redis_val: i64,
         remote_deltas: i64,
-        redis_ttl_ms: i64,
-        ttl_margin: Duration,
+        redis_expiry: i64,
     ) -> Arc<CachedCounterValue> {
-        let cache_ttl = self.ttl_from_redis_ttl(
-            redis_ttl_ms,
-            counter.seconds(),
-            redis_val,
-            counter.max_value(),
-        );
-        if let Some(ttl) = cache_ttl.checked_sub(ttl_margin) {
-            if ttl > Duration::ZERO {
+        if redis_expiry > 0 {
+            let expiry_ts = SystemTime::UNIX_EPOCH + Duration::from_micros(redis_expiry as u64);
```

Comment:
According to the docs for [PEXPIRETIME](https://redis.io/docs/latest/commands/pexpiretime/) it's returned as millis, not micros so I see the `if expiry_ts > SytemTime::now()` returning false when it's supposed to be true

```suggestion
            let expiry_ts = SystemTime::UNIX_EPOCH + Duration::from_millis(redis_expiry as u64);
```

Reply:
Scares me this didn't create any test failures! 🤦

Reply:
Ah! Because I had written the tests to inject time in micros too… fixed that part as well.